### PR TITLE
[feature](external) process tbl/db exist when create/drop db/tbl

### PR DESCRIPTION
**Pull request by wsjz**
## Proposed changes

Issue Number: https://github.com/apache/doris/issues/31442

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Review by wuwenchi — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetadataOps.java [hunk 1]
```diff
@@ -186,24 +212,22 @@ public void createTable(CreateTableStmt stmt) throws UserException {
...
-    }
-
     @Override
     public void dropTable(DropTableStmt stmt) throws DdlException {
         String dbName = stmt.getDbName();
         ExternalDatabase<?> db = catalog.getDbNullable(stmt.getDbName());
         if (db == null) {
             throw new DdlException("Failed to get database: '" + dbName + "' in catalog: " + catalog.getName());
         }
+        ExternalTable table = db.getTableNullable(stmt.getTableName());
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetadataOps.java:222]: I think we should check whether table exist in hms, not just in Doris.
Same as `databaseExist()`

File: fe/fe-core/src/main/java/org/apache/doris/datasource/iceberg/IcebergMetadataOps.java [hunk 2]
```diff
@@ -109,6 +132,14 @@ public void createTable(CreateTableStmt stmt) throws UserException {
             throw new UserException("Failed to get database: '" + dbName + "' in catalog: " + dorisCatalog.getName());
         }
         String tableName = stmt.getTableName();
+        if (db.getTable(tableName).isPresent()) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/iceberg/IcebergMetadataOps.java:135]: Same comment as hive catalog

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

**Review by kaka11chen — APPROVED** [review]
LGTM

**Comment by morningman** [comment]
run buildall

**Review by morningman — APPROVED** [review]

**Pull request merged**